packages/relay/src/lib/model.ts

```diff
@@ -103,36 +104,36 @@ export class CachedBlock extends Block {
 
 export class Receipt {
     public readonly transactionHash:string;
-    public readonly transactionIndex:number;
+    public readonly transactionIndex:string;
     public readonly blockHash:string;
-    public readonly blockNumber:number;
+    public readonly blockNumber:string;
     public readonly from:string;
     public readonly to:(undefined|string);
-    public readonly cumulativeGasUsed:number;
-    public readonly gasUsed:number;
+    public readonly cumulativeGasUsed:string;
+    public readonly gasUsed:string;
     public readonly contractAddress:(undefined|string);
-    public readonly logs:string[];
+    public readonly logs:Log[];
     public readonly logsBloom:string;
     public readonly root:(undefined|string);
-    public readonly status:(undefined|number);
-    public readonly effectiveGasPrice:(undefined|number);
+    public readonly status:(undefined|string);
+    public readonly effectiveGasPrice:(undefined|string);
 
     constructor(txHash:string, record:TransactionRecord, block:Block) {
-        const gasUsed = record.contractFunctionResult == null ? 0 : record.contractFunctionResult.gasUsed;
+        const gasUsed = EthImpl.numberTo0x(record.contractFunctionResult == null ? 0 : record.contractFunctionResult.gasUsed.toNumber());
         const contractAddress = record.contractFunctionResult == undefined ? undefined : "0x" + record.contractFunctionResult.contractId?.toSolidityAddress();
 
         this.transactionHash = txHash;
-        this.transactionIndex = 0;
-        this.blockNumber = Number(block.number);
+        this.transactionIndex = "0x0";
+        this.blockNumber = block.number;
         this.blockHash = block.hash;
         this.from = '0x';
         // TODO this.to = record.contractFunctionResult?.contractId;
-        this.cumulativeGasUsed = Number(gasUsed);
-        this.gasUsed = Number(gasUsed);
+        this.cumulativeGasUsed = gasUsed;
+        this.gasUsed = gasUsed;
         this.contractAddress = contractAddress;
         this.logs = [];
         this.logsBloom = '';
-        this.status = record.receipt.status == Status.Success ? 1 : 0;
+        this.status = record.receipt.status == Status.Success ? "0x1" : "0x0";
```

Comment:
Also move `0x1` to a static class variable
```suggestion
        this.status = record.receipt.status == Status.Success ? "0x1" : EthImpl.zeroHex;
```

Reply:
I opted for a direct rollback on this line.

Reply:
Gotcha. `0x0` and `0x1` should still be static class constants

Reply:
NO biggie. We can clean this and other files up later